src/lightgbm/src/main/scala/LightGBMUtils.scala

```diff
@@ -137,24 +212,24 @@ object LightGBMUtils {
     * @param defaultListenPort The default listening port.
     * @return The list of nodes in host:port format.
     */
-  def getNodesFromPartitionsLocal(processedData: DataFrame, defaultListenPort: Int): (String, Int) = {
+  def getNodesFromPartitionsLocal(processedData: DataFrame,
+                                  defaultListenPort: Int): Array[(Int, String)] = {
     import processedData.sparkSession.implicits._
     val blockManager = BlockManagerUtils.getBlockManager(processedData)
     val host = blockManager.master.getMemoryStatus.flatMap({ case (blockManagerId, _) =>
       Some(blockManagerId.host)
     }).head
-    val nodes = reduceNodesToString(
+    val nodes =
       processedData.mapPartitions((_: Iterator[Row]) => {
         // The logic below is to get it to run in local[*] spark context
         val id = getId()
-        Array(host + ":" + (defaultListenPort + id)).toIterator
-      })
-    )
-    (nodes, nodes.split(",").length)
+        Array((id, host)).toIterator
+      }).collect()
+    nodes
   }
 
-  private def reduceNodesToString(nodes: Dataset[String]): String = {
-    nodes.collect().sorted.distinct.reduceLeft((val1, val2) => val1 + "," + val2)
+  private def reduceNodes(nodes: Dataset[(String, Int)]): Array[(String, Int)] = {
```

Comment:
probably dont need this

Reply:
removed - apparently the method is not used, looks like a typo